src/snmalloc/mem/freelist.h

```diff
@@ -688,7 +678,7 @@ namespace snmalloc
       /**
        * Set the builder to a not building state.
        */
-      constexpr void init()
+      constexpr void init(void* slab, const FreeListKey& key)
```

Comment:
Given `useless_ptr_from_addr`, can this be
```suggestion
      constexpr void init(address_t slab, const FreeListKey& key)
```

If so, I think `extract_segment` and `initialise` and `initialise_large` (in `metadata.h`) can also take `address_t`.  I think the update propagation stops at their callers.